fido/pronom/soap.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+FIDO: Format Identifier for Digital Objects
+
+Copyright 2010 The Open Preservation Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+PRONOM format signatures SOAP calls.
+"""
+import sys
+import urllib.request
+import xml.etree.ElementTree as ET
+
+from fido import __version__
+ENCODING = 'utf-8'
+XML_PROC = '<?xml version="1.0" encoding="{}"?>'.format(ENCODING)
+TNA_DOMAIN = 'nationalarchives.gov.uk'
+PRONOM_HOST = 'www.{}'.format(TNA_DOMAIN)
+PRONOM_NS = 'http://pronom.{}'.format(TNA_DOMAIN)
+SIG_NS = 'http://{}/pronom/SignatureFile'.format(PRONOM_HOST)
+
+NS = {
+    'soap': 'http://schemas.xmlsoap.org/soap/envelope/',
+    'xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+    'xsd': 'http://www.w3.org/2001/XMLSchema',
+    'pronom': PRONOM_NS,
+    'sig': SIG_NS
+}
+
+HEADERS = {
+    'Host': PRONOM_HOST,
+    'User-Agent': 'PRONOM UTILS v{0} (OPF)'.format(__version__),
+    'Content-type': 'text/xml; charset="UTF-8"'
+}
+
+def get_pronom_sig_version():
+    """
+    Get PRONOM signature version.
+
+    Return latest signature file version number as an int.
+    Raises an HTTPError if there are problems.
+    """
+    try:
+        tree = _get_soap_ele_tree('getSignatureFileVersionV1')
+        ver_ele = tree.find('.//pronom:Version/pronom:Version', NS)
+        return int(ver_ele.text)
+    except Exception as e:
+        sys.stderr.write('get_pronom_sig_version(): unknown error: {}'.format(str(e)))
+        raise e
+
+def get_pronom_signature():
+    """
+    Get PRONOM signature.
+
+    Return a tuple comprising the latest signature XML file as string and a count
+    of the FileFormat elements contained as an integer.
+    Upon error, write to `stderr` and return the tuple [], False.
+    """
+    try:
+        tree = _get_soap_ele_tree('getSignatureFileV1')
+        for prefix, uri in NS.items():
+            ET.register_namespace(prefix, uri)
+        sigfile_ele = tree.find('.//pronom:SignatureFile', NS)
+        format_ele_len = len(sigfile_ele.findall('.//sig:FileFormat', NS))
+        if format_ele_len < 1:
+            sys.stderr.write("get_pronom_signature(): could not parse XML from SOAP response: file")
+            return [], False
+        return '<?xml version="1.0" encoding="UTF-8"?>\n' + ET.tostring(sigfile_ele, encoding='unicode'), format_ele_len
```

Comment:
This is the last bit that seems to be incompatible with Python 2.7. Not sure what would be the best solution, maybe something like the following:

```python
from __future__ import unicode_literals
from six import text_type
# ...
return '<?xml version="1.0" encoding="UTF-8"?>\n' + text_type(ET.tostring(sigfile_ele, encoding='utf-8')), format_ele_len
```

Reply:
Again if we decide to revert the sig prep stuff to 2.7 then will fix it.